app/src/main/java/com/infomaniak/mail/utils/MessageUtils.kt

```diff
@@ -50,5 +50,14 @@ object MessageUtils {
     }
 }
 
-data class ThreadMessageToExecuteAction(val thread: Thread, val messageUid: String)
+sealed class ThreadMessageInteraction(open val messageUid: String) {
+    data class Action(val thread: Thread, override val messageUid: String) : ThreadMessageInteraction(messageUid)
+    data class Reaction(override val messageUid: String): ThreadMessageInteraction(messageUid)
+}
+
+data class ThreadMessageToExecuteInteraction(
+    val thread: Thread,
+    val messageUidToExecuteAction: String,
+    val messageUidToExecuteReaction: String?
+)
```

Comment:
The [comment](https://github.com/Infomaniak/android-kMail/pull/2602#discussion_r2381925053) about new lines between definitions of classes is still applicable here